### PR TITLE
prioritize critical css : avoid adding duplicate ids 

### LINKED ISSUE #1878: prioritize critical css need to remove id attributes
Hi
Using the prioritize_critial_css inlined styles with id attibutes get duplicate id´s, one when the filter restore the inlined style (if it is in the critical path) and other when the original inlined style is loaded on the onload event.

### PR DESCRIPTION
**Pull request by Lofesa**
This is the same as PR #1886 , closed when the branch is removed.
PR #1886  was awaiting a request for review by @jmarantz

### CODE REVIEW

**Comment by Lofesa** [comment]
Close and reopen to run CI again

**Comment by oschaaf** [comment]
Cross linking issue this should close: https://github.com/apache/incubator-pagespeed-mod/issues/1878

**Review by oschaaf — CHANGES_REQUESTED** [review]
@Lofesa the change makes sense to me, however, I think we also need to consider the child elements of the child elements, and so on. So we should recursively strip id attributes. Would you be up for making that change?

File: net/instaweb/rewriter/critical_css_loader.js [hunk 1]
```diff
@@ -39,11 +39,16 @@ pagespeed.CriticalCssLoader.addAllStyles = function() {
...
   for (var i = 0, e; e = elements[i]; ++i) {
     if (e.nodeName != 'NOSCRIPT') { continue; }
     var div = document.createElement('div');
     div.innerHTML = e.textContent;
-    document.body.appendChild(div);
+    var children = div.childNodes;
+    for (var v = 0; v < children.length; ++v) {
+      children[v].removeAttribute('id');
+    }  
+  document.body.appendChild(div);
```
Comment (oschaaf) [inline net/instaweb/rewriter/critical_css_loader.js]: nit: indent

Comment (Lofesa) [reply net/instaweb/rewriter/critical_css_loader.js]: The change you requested is to indent the line
  `document.body.appendChild(div);` ?

Comment (oschaaf) [reply net/instaweb/rewriter/critical_css_loader.js]: yes, that would be great. But we may also want to recurse? So not only remove the `id` attribute from the direct children of the element, but also the children of the children, and so on?

Comment (Lofesa) [reply net/instaweb/rewriter/critical_css_loader.js]: But childNode don´t get all childs and sub-childs nodes?

Comment (oschaaf) [reply net/instaweb/rewriter/critical_css_loader.js]: Oh in that case you’re right this is OK.

Comment (oschaaf) [reply net/instaweb/rewriter/critical_css_loader.js]: Sorry about the noise!

**Review by oschaaf — APPROVED** [review]

**Pull request merged**